### DISCUSSION TITLE
Add classes from database during server-side page render in Next JS

### DISCUSSION THREAD

**Discussion by dbondarchuk** [tailwindlabs/tailwindcss#14186] [Help]
Hi
I have a question regarding using Tailwind in a hybrid postcss / runtime environment
I have a Next.JS with postcss configured. The app allows user to have their custom page templates done in mdx to be stored in DB
I would love to be able to add the extracted tailwind-based utilities to the layout.css during the page server build.
Is it possible?
Currently, my workaround is to additionally load CDN tailwind in runtime, which is not the most ideal solution.

**Comment by wongjn** [comment]
You could try using the PostCSS API to programmatically run Tailwind:

```js
const postcss = require('postcss');
const tailwindcss require('tailwindcss');
const baseConfig = require('./tailwind.config');

const sourceCSS = '@tailwind base; @tailwind components; @tailwind utilities';
const config = {
  presets: [baseConfig],
  content: [{ raw: markup }],
};
postcss([tailwindcss(config)])
  .process(sourceCSS)
  .then((css) => {
    // `css` is a string of the compiled CSS.
  });
```

I'm not sure how this would integrate into Next.js in your project, but could be a starting point.
